### ISSUE TITLE
first and last words don't have timings issue - edge case

### ISSUE THREAD

**Issue by pietrop** [bbc/stt-align-node#13]
possible fix via @Laurian

>basically I pad the word array to be sure the 1st and last word has timing, otherwise it will extrapolate timing on words that lack timing and are on the edge, getting by that timings outside the paragraph boundaries (even negative timings on 1st paragraph)
```js
words = [
      {
        text: 'STARTSTART',
        start: words[0].start - (0.08475 + 0.05379 * 'STARTSTART'.length),
        end: words[0].start,
      },
      ...words,
      {
        text: 'ENDEND',
        start: words[words.length - 1].end,
        end: words[words.length - 1].end + 0.08475 + 0.05379 * 'ENDEND'.length,
      },
    ];

    const resultAligned = alignSTT(
      {
        words,
      },
      `STARTSTART ${text} ENDEND`
    );
```

**Comment by Laurian** [comment]
For example missing timings at the beginning 
```js
"words": [
        {
          "text": "there"
        },
        {
          "text": "is"
        },
        {
          "text": "a"
        },
        {
          "text": "day"
        },
        {
          "text": "about"
        },
        {
          "start": 14.13,
          "end": 14.37,
          "text": "ten"
        },
        {
          "start": 14.37,
          "end": 14.61,
          "text": "years"
        },
        {
          "start": 14.61,
          "end": 15.15,
          "text": "ago"
        },
```

will have `everpolate` extrapolate them, and if you align segment by segment you will get timings that might overlap the previous segment timing, sometimes even negative timings

This paragraph should start at 13.05, but there is no hard stop for start/end of the alignment
```js
 {
    "text": "There",
    "start": 12.930000000000009,
    "end": 13.169999999999998
  },
  {
    "text": "was",
    "start": 13.170000000000007,
    "end": 13.41
  },
  {
    "end": 13.650000000000004,
    "start": 13.410000000000005,
    "text": "a"
  },
  {
    "end": 13.890000000000002,
    "start": 13.650000000000004,
    "text": "day,"
  },
  {
    "end": 14.13,
    "start": 13.890000000000002,
    "text": "about"
  },
  {
    "end": 14.37,
    "start": 14.13,
    "text": "10"
  },
```

**Comment by Laurian** [comment]
I needed this (#14) for the case when I realign paragraph by paragraph during editing and I need to guard the extrapolated timings not to overlap the previous and next paragraph timings.

**Comment by Laurian** [comment]
Fixed with #14
